### PR TITLE
vault password is renewed every 7 days

### PR DESCRIPTION
**Pull request by CChemin**
if the vault password used in services is older than 7 days, onyxia-ui asks for its renewal. It is still possible to get older passwords (in a limit of 10). These older passwords can be browsed in the account interface.

### CODE REVIEW

File: src/js/vault-client/vault-api.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ import conf from '../configuration';
 
 const VAULT_BASE_URI = conf.VAULT.VAULT_BASE_URI;
 const VAULT_KV_ENGINE = conf.VAULT.VAULT_KV_ENGINE;
+const NB_DAYS_BEFORE_PWD_RENEWAL = 7;
```
Comment (olevitt) [inline src/js/vault-client/vault-api.js:10]: I suggest we move this to an environment variable. People may want to disable / tune this behaviour on their installation

File: src/js/vault-client/vault-api.js [hunk 2]
```diff
@@ -58,12 +59,18 @@ const fetchVaultToken = async () => {
...
-		.then(({ data: { data : {data}} }) => store.dispatch(newVaultData(data)))
+		.then(({ data : {data} }) => 
+			pwdMustBeRenewed(data.metadata.created_time) ? resetVaultPwd(idep) : store.dispatch(newVaultData(data.data))
+			)
 		.catch(() => {
 			resetVaultPwd(idep);
 		});
 };
 
+const pwdMustBeRenewed = (created_time) => {
```
Comment (olevitt) [inline src/js/vault-client/vault-api.js:70]: I dislike the `snake_case` convention. But that may be personal preference. Anyway, onyxia is mostly using `lowerCamelCase` at this point, I suggest we try to follow this convention as much as possible.

File: src/js/vault-client/vault-api.js [hunk 3]
```diff
@@ -58,12 +59,18 @@ const fetchVaultToken = async () => {
...
+	return Date.now() - Date.parse(created_time) > 8400*NB_DAYS_BEFORE_PWD_RENEWAL
```
Comment (olevitt) [inline src/js/vault-client/vault-api.js:71]: I may be missing something here, what is `8400` equivalent to ? Is is supposed to `86400` (number of seconds in a day) ?

**Comment by garronej** [comment]
@CChemin Thank you for your input, however, the codebase has dramatically changed since this PR was open. I keep this feature in the back of my mind, though.

**Pull request closed without merging**